### DISCUSSION TITLE
How to use HOC in React?

### DISCUSSION THREAD

**Discussion by starlitnightsky** [starlitnightsky/Real-Estate#1] [Announcements]
Hello, every one.
How to use HOC in React?

**Comment by wannacfuture** [comment]
It's simple.
function withData(WrappedComponent) {
  return class extends React.Component {
    state = {
      data: [],
    };

    componentDidMount() {
      // fetch data and update state
      fetch("https://example.com/data")
        .then((response) => response.json())
        .then((data) => this.setState({ data }));
    }

    render() {
      return <WrappedComponent data={this.state.data} {...this.props} />;
    }
  };
}

// Use the Higher-Order Component
class MyComponent extends React.Component {
  render() {
    const { data } = this.props;
    return (
      <div>
        {data.map((item) => (
          <p>{item.title}</p>
        ))}
      </div>
    );
  }
}

export default withData(MyComponent);
